inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -210,6 +210,14 @@ def is_submission(obj, eng):
 
 
 @with_debug_logging
+def add_holdingpen_entry_id(obj, eng):
+    """Add holdingpen entry id for this record in 'submission_number' """
+    method = get_value(obj.data, 'acquisition_source.method', default='')
+    if method == 'hepcrawl':
+        obj.data['acquisition_source']['submission_number'] = obj.id
```

Comment:
I am not sure: don't we always want `obj.id` here?

Reply:
You are right, this method is being called only for "hepcrawl" records anyway, this is a bit of a redundant check. I will fix it!

Reply:
Mmh. So there are two paths that are setting `obj.id` independently?